src/modules/powerrename/lib/trace.cpp

```diff
@@ -71,3 +72,29 @@ void Trace::RenameOperation(_In_ UINT totalItemCount, _In_ UINT selectedItemCoun
         TraceLoggingWideString(extensionList, "ExtensionList"));
 }
 
+void Trace::SettingsChanged() noexcept
+{
+    wchar_t searchBuffer[CSettings::MAX_INPUT_STRING_LEN];
+    searchBuffer[0] = L'\0';
+    CSettings::GetSearchText(searchBuffer, ARRAYSIZE(searchBuffer));
+
+    wchar_t replaceBuffer[CSettings::MAX_INPUT_STRING_LEN];
+    replaceBuffer[0] = L'\0';
+    CSettings::GetReplaceText(replaceBuffer, ARRAYSIZE(replaceBuffer));
+
+    TraceLoggingWrite(
+        g_hProvider,
+        "PowerRename_SettingsChanged",
+        ProjectTelemetryPrivacyDataTag(ProjectTelemetryTag_ProductAndServicePerformance),
+        TraceLoggingKeyword(PROJECT_KEYWORD_MEASURE),
+        TraceLoggingBoolean(CSettings::GetEnabled(), "IsEnabled"),
+        TraceLoggingBoolean(CSettings::GetShowIconOnMenu(), "ShowIconOnMenu"),
+        TraceLoggingBoolean(CSettings::GetExtendedContextMenuOnly(), "ExtendedContextMenuOnly"),
+        TraceLoggingBoolean(CSettings::GetPersistState(), "PersistState"),
+        TraceLoggingBoolean(CSettings::GetMRUEnabled(), "IsMRUEnabled"),
+        TraceLoggingUInt64(CSettings::GetMaxMRUSize(), "MaxMRUSize"),
+        TraceLoggingUInt64(CSettings::GetFlags(), "Flags"),
+        TraceLoggingWideString(searchBuffer, "SearchText"),
+        TraceLoggingWideString(replaceBuffer, "ReplaceText")
```

Comment:
Please remove `SearchText` and `ReplaceText` since that is personal information that can't be traced.